src/runtime/manifest.ts

```diff
@@ -1315,13 +1317,26 @@ ${e.message}
   registerStore(store: UnifiedStore, tags: string[]): void {
     // Only register stores that have non-volatile storage key and don't have a
     // #volatile tag.
-    if (!this.findStoreById(store.id) &&
-        (store.storageKey as StorageKey).protocol !== 'volatile' &&
-        !tags.includes('volatile')) {
+    if (!this.findStoreById(store.id) && !this.isVolatileStore(store, tags)) {
+      if (Flags.defaultReferenceMode && !(store.storageKey instanceof ReferenceModeStorageKey)) {
+        // Note: skip adding backing and container stores to context, if using reference-mode stores.
+        // TODO: there must be a better way!
+        return;
+      }
```

Comment:
why is this here? What's trying to add the backing & container stores? I know we add them to the arc to make sure serialization happens properly - is that being somehow copied into the manifest?